llama_index/readers/file/base.py

```diff
@@ -186,34 +189,41 @@ def load_data(self, concatenate: bool = False) -> List[Document]:
                 with open(input_file, "r", errors=self.errors, encoding="utf8") as f:
                     data = f.read()
 
-            metadata: Optional[dict] = None
-            if self.file_metadata is not None:
-                metadata = self.file_metadata(str(input_file))
+            file_metadata: Optional[dict] = None
+            if self.file_metadata_fn is not None:
+                file_metadata = self.file_metadata_fn(str(input_file))
 
             if isinstance(data, ImageParserOutput):
                 # process image
                 image_docs.append(
-                    ImageDocument(text=data.text, extra_info=metadata, image=data.image)
+                    ImageDocument(
+                        text=data.text, extra_info=file_metadata, image=data.image
+                    )
                 )
             elif isinstance(data, List):
-                # process list of str
-                data_list.extend(data)
-                repeated_metadata: List[Optional[dict]] = [
-                    deepcopy(metadata) for _ in range(len(data))
-                ]
-                metadata_list.extend(repeated_metadata)
+                for item in data:
```

Comment:
Hmm is it possible to keep consistent with the implementation on llama-hub where instead of parsers each file reader just returns its own set of documents? 

Obviously it's not great there's an inconsistency between the implementation in this repo vs llama-hub but would prefer to be more consistent with the llama-hub one if possible

Reply:
Created https://github.com/jerryjliu/llama_index/pull/3352 and rebased on top of it.